src/transport/validator/uattributesvalidator.rs

```diff
@@ -25,169 +26,177 @@ use std::time::SystemTime;
-            if !UuidUtils::is_valid_uuid(&reqid) {
-                return UStatus::fail_with_msg_and_reason(
-                    "Invalid correlation UUID",
-                    UCode::InvalidArgument,
-                );
-            }
+    /// Returns a `ValidationResult` that is success or failed with a failure message.
+    fn validate_sink(&self, attributes: &UAttributes) -> ValidationResult {
+        if let Some(sink) = &attributes.sink {
+            return UriValidator::validate(sink);
         }
-        UStatus::ok()
+        ValidationResult::Success
     }
 
-    /// Validate the time to live configuration. If the UAttributes does not contain a time to live then the UStatus is ok.
+    /// Validates the permission level for the default case. If the UAttributes does not contain
+    /// a permission level then the ValidationResult is ok.
     ///
     /// # Arguments
     ///
-    /// * `attributes` - `UAttributes` object containing the message priority to validate.
+    /// * `attributes` - `UAttributes` object containing the permission level to validate.
     ///
     /// # Returns
     ///
-    /// Returns a `UStatus` that is success or failed with a failure message.
-    fn validate_ttl(&self, attributes: &UAttributes) -> UStatus {
-        if let Some(ttl) = attributes.ttl {
-            if ttl < 1 {
-                return UStatus::fail_with_msg_and_reason("Invalid TTL", UCode::InvalidArgument);
+    /// Returns a `ValidationResult` that is success or failed with a failure message.
+    fn validate_permission_level(&self, attributes: &UAttributes) -> ValidationResult {
+        if let Some(plevel) = &attributes.permission_level {
+            if *plevel < 1 {
+                return ValidationResult::Failure("Invalid Permission Level".to_string());
             }
         }
-        UStatus::ok()
+        ValidationResult::Success
     }
 
-    /// Validates the sink URI for the default case.
+    /// Validates the communication status (`commStatus`) for the default case. If the UAttributes
+    /// does not contain a request id then the UStatus is ok.
     ///
     /// # Arguments
     ///
-    /// * `attributes` - `UAttributes` object containing the sink to validate.
+    /// * `attributes` - `UAttributes` object containing the communication status to validate.
     ///
     /// # Returns
     ///
-    /// Returns a `UStatus` indicating whether the URI is valid or not.
-    fn validate_sink(&self, attributes: &UAttributes) -> UStatus {
-        if let Some(sink) = &attributes.sink {
-            return UriValidator::validate(sink);
+    /// Returns a `ValidationResult` that is success or failed with a failure message.
+    fn validate_commstatus(&self, attributes: &UAttributes) -> ValidationResult {
+        if let Some(cs) = attributes.commstatus {
+            if UCode::try_from(cs).is_err() {
+                return ValidationResult::Failure("Invalid Communication Status Code".into());
```

Comment:
Would it be possible to include the `unknown` status code as part of the message error?

Reply:
done